test/e2e/case31_policy_history_test.go

```diff
@@ -0,0 +1,153 @@
+	case31ConfigPolicy           = "../resources/case31_policy_history/pod-config-policy.yaml"
+	case31PolicyName             = "test-policy-security"
+	case31ConfigPolicyName       = "config-policy-pod"
+	case31PolicyNumber           = "../resources/case31_policy_history/pod-policy-number.yaml"
+	case31ConfigPolicyNumber     = "../resources/case31_policy_history/pod-config-policy-number.yaml"
+	case31PolicyNumberName       = "test-policy-security-number"
+	case31ConfigPolicyNumberName = "config-policy-pod-number"
+)
+
+var _ = Describe("Test policy history message when KubeAPI return "+
+	"omits values in the returned object", Ordered, func() {
+	Describe("status toggling should not be generated When Policy include default value,", Ordered, func() {
+		It("creates the policyconfiguration "+case31Policy, func() {
+			utils.Kubectl("apply", "-f", case31Policy, "-n", "managed")
+		})
+
+		It("verifies the policy "+case31PolicyName+" in "+testNamespace, func() {
+			By("bind policy and configurationpolicy")
+			parent := utils.GetWithTimeout(clientManagedDynamic, gvrPolicy,
+				case31PolicyName, testNamespace, true, defaultTimeoutSeconds)
+			Expect(parent).NotTo(BeNil())
+			// biding config
+			plcDef := utils.ParseYaml(case31ConfigPolicy)
+			ownerRefs := plcDef.GetOwnerReferences()
+			ownerRefs[0].UID = parent.GetUID()
+			plcDef.SetOwnerReferences(ownerRefs)
+			_, err := clientManagedDynamic.Resource(gvrConfigPolicy).Namespace(testNamespace).
+				Create(context.TODO(), plcDef, metav1.CreateOptions{})
+			Expect(err).To(BeNil())
+
+			By("check configurationpolicy exist")
+			plc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+				case31ConfigPolicyName, testNamespace, true, defaultTimeoutSeconds)
+			Expect(plc).NotTo(BeNil())
+		})
+
+		It("check history toggling", func() {
+			By("wait until pod is up")
+			Eventually(func() interface{} {
+				managedPlc := utils.GetWithTimeout(clientManagedDynamic, gvrConfigPolicy,
+					case31ConfigPolicyName, testNamespace, true, defaultTimeoutSeconds)
+
+				return utils.GetComplianceState(managedPlc)
+			}, defaultTimeoutSeconds, 1).Should(Equal("Compliant"))
+
+			By("Wait for few second to check toggling")
+			utils.Pause(7)
+			Consistently(func() int {
+				eventlen := len(utils.GetMatchingEvents(clientManaged, testNamespace,
+					case31ConfigPolicyName, case31ConfigPolicyName, "NonCompliant;", defaultTimeoutSeconds))
+
+				return eventlen
+			}, 30, 5).Should(BeNumerically("<", 2))
+
+			// NOTE: pick policy event, these event's reason include ConfigPolicyName
+			Consistently(func() int {
+				eventlen := len(utils.GetMatchingEvents(clientManaged, testNamespace,
+					case31PolicyName, case31ConfigPolicyName, "NonCompliant;", defaultTimeoutSeconds))
+
+				return eventlen
+			}, 30, 5).Should(BeNumerically("<", 2))
+		})
+		AfterAll(func() {
+			deleteConfigPolicies([]string{case31PolicyName})
+			utils.Kubectl("delete", "event", "--all", "-n", "managed")
```

Comment:
Could you please scope this tighter to the policy like using a `involvedObject` filter? Way down on my to do list is to make the config-policy-controller tests run in parallel to make it faster and this would take us a step back from being able to do that.